benches/benches/agui_core/tree.rs

```diff
@@ -84,15 +93,20 @@ fn tree_iter_up(c: &mut Criterion) {
                 tree
             },
             |tree| {
-                for _ in 0..1000 {
-                    let mut walker = tree.iter_up();
+                // wrap the iterator in an Rc<RefCell<_>>
+                let walker = Rc::new(RefCell::new(tree.iter_up())); 
 
-                    while black_box(walker.next().is_some()) {}
+                for _ in 0..1000 {
+                    // borrow mutably from the Rc<RefCell<_>>
+                    let mut walker = walker.borrow_mut(); 
+                    // use while let loop to retrieve and check value
+                    while let Some(_) = walker.next() {} 
```

Comment:
Wouldn't this effectively iterate up the tree once, since `iter_up()` is only called a single time? Effectively this would walk up the tree a single time, then no-op for the remaining 999 loops

Reply:
It should loop through 1000 times, and the do the ops for each loop due to the for loops. Unless I'm missing something?

Reply:
Sorry for the late response,  this got lost in my notifications. Calling `iter_up()` once creates the iterator a single time. Once it's exhausted in the first run of the for loop, it no-ops from there.